Add "Media Updates in Chrome 61" /updates article

beaufortfrancois opened the pull request:
Hello @mounirlamouri @dalecurtis,
Let me know what you think of this draft.

Live preview at https://pr-4857-dot-web-central.appspot.com/web/updates/2017/07/chrome-61-media-updates

mounirlamouri commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,92 @@
+
+This magic behaviour only happens when:
+
+- device is an Android phone
```
underline not tablet?

beaufortfrancois replied:
Done.

mounirlamouri commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,92 @@
+- video is currently playing
+- at least 75% of the video is visible (on-screen)
+- orientation rotates by 90 degrees (not 180 degrees)
+- there is no fullscreen element yet
```
Worth pointing out that if the page has a screen lock, it does not happen too.

beaufortfrancois replied:
Done.

mounirlamouri commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,92 @@
+- at least 75% of the video is visible (on-screen)
+- orientation rotates by 90 degrees (not 180 degrees)
+- there is no fullscreen element yet
+
```
Should you remind developers that they can implement this themselves? I believe you mention this in your best practice article, maybe point to it?

beaufortfrancois replied:
Done.

mounirlamouri commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,92 @@
+      if (document.hidden) {
+        // Disable video track when page is hidden.
+        selectedVideoTrackIndex = video.videoTracks.selectedIndex;
+        video.videoTracks[selectedVideoTrackIndex].selected = false;
```
that's behind a flag, right?

beaufortfrancois replied:
It is for now yes but this code is meant to explain what is going on. WDYT?

mounirlamouri commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,92 @@
+        // Re-enable video track when page is not hidden anymore.
+        video.videoTracks[selectedVideoTrackIndex].selected = true;
+      }
+    });
```
Should we suggest that developers reduce the quality of the streem if they think it's worth it?

beaufortfrancois replied:
Done.

beaufortfrancois commented:
@mounirlamouri Your feedback has been addressed at https://github.com/google/WebFundamentals/pull/4857/commits/ee832e1fe0878ddd50929772b35d6dd97f674b6e. Thanks!

mounirlamouri commented:
lgtm :+1:

jondahlke commented:
LGTM

dalecurtis commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,102 @@
+{# wf_published_on: 2017-07-17 #}
+{# wf_tags: news,chrome61,media #}
+{# wf_featured_image: /web/updates/images/generic/animations.png #}
+{# wf_featured_snippet: Background video track optimization and automatic video fullscreen when device is rotating features are there! #}
```
"Background video track optimizations and rotate-for-fullscreen are here!" ?

dalecurtis commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,102 @@
+
+- Chrome now [disables video tracks when a MSE video is played in the
+  background](#background-video-track-optimization) to optimize performance. 
+- A video will [automatically go fullscreen when device is
```
s/A video/Video/
s/automatically//

beaufortfrancois replied:
Done.

dalecurtis commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,102 @@
+- Chrome now [disables video tracks when a MSE video is played in the
+  background](#background-video-track-optimization) to optimize performance. 
+- A video will [automatically go fullscreen when device is
+  rotating](#auto-fullscreen-rotate).
```
s/rotating/rotated/

beaufortfrancois replied:
Done.

dalecurtis commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,102 @@
+## Background video track optimization (MSE only) {: #background-video-track-optimization}
+
+To improve battery life, Chrome now disables video tracks when the video is
+played in the background if this one uses [Media Source Extensions (MSE)].
```
s/if this one uses/if the video uses/

beaufortfrancois replied:
Done.

dalecurtis commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,102 @@
+
+[Chromium Bug](https://bugs.chromium.org/p/chromium/issues/detail?id=663999)
+
+## Automatic video fullscreen when device is rotating {: #auto-fullscreen-rotate }
```
Rotate-for-fullscreen ? s/rotating/rotated/ at least

beaufortfrancois commented:
Thank you @dalecurtis! I've addressed your feedback.
I'm still using "Automatic video fullscreen when device is rotated" though as "rotate-for-fullscreen" seems cryptic at first glance in my opinion. If you feel strongly, we may want to revisit my decision.

beaufortfrancois commented:
@jpmedley You may review this article now. Thanks!

beaufortfrancois commented:
(gentle ping) @jpmedley

jpmedley commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,101 @@
+## Automatic video fullscreen when device is rotated {: #auto-fullscreen-rotate }
+
+When a video is playing in the viewport, rotating the device to landscape
+orientation will now fullscreen the video. Rotating the device to portrait puts
```
This sentence is a little clunky. Try:

If you rotate a device to landscape while a video is playing, playback will automatically switch to fullscreen mode.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,101 @@
+
+When a video is playing in the viewport, rotating the device to landscape
+orientation will now fullscreen the video. Rotating the device to portrait puts
+the video back to windowed mode as you would expect.
```
Delete 'as you would expect.' It doesn't add any information, and I'm not sure that I would expect this.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,101 @@
+orientation will now fullscreen the video. Rotating the device to portrait puts
+the video back to windowed mode as you would expect.
+
+Note that you can implement manually this behaviour yourself (see [Mobile Web Video
```
Change 'implement manually' to 'manually implement'.

jpmedley commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,101 @@
+- video is currently playing
+- at least 75% of the video is visible (on-screen)
+- orientation rotates by 90 degrees (not 180 degrees)
+- there is no fullscreen element yet
```
I didn't know what a fullscreen element was. Link to this: https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenElement

Or if you know of something more appropriate, use that.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/2017/07/chrome-61-media-updates.md:
```diff
@@ -0,0 +1,101 @@
+- at least 75% of the video is visible (on-screen)
+- orientation rotates by 90 degrees (not 180 degrees)
+- there is no fullscreen element yet
+- screen is not locked using the Screen Orientation API
```
Maybe consider linking to the spec. https://w3c.github.io/screen-orientation/

The API isn't documented on MDN. It looks like it shipped in the last version before we started contributing to MDN :(

beaufortfrancois replied:
Done.

beaufortfrancois commented:
@jpmedley Thank you! I've addressed your feedback and merged back your changes.
